### PR TITLE
[DO NOT MERGE] Repro weirdness with the Inception test - weight init

### PR DESCRIPTION
**Pull request by prigoyal**
while e2e testing the non-pretrained model, we found that the existing weight init for Conv layers in Inception leads to non-matching output with caffe2. However, if we change the weight init to 
'm.weight.data.normal_(0.0, 0.02)' then it works fine and catch output with caffe2. This is a very weird behavior.

We also tried: 
weight init = constant fill 0, 1, 1e5 and they all work fine

@apaszke @soumith have you seen this behaviour before? To repro, cherry-pick this commit on hash 6747cbd and run 

python test/model_defs/caffe2_pytorch_test_models.py TestCaffe2BackendEmbed.test_inception

### CODE REVIEW

**Comment by ezyang** [comment]
From Chat:

INCEPTION: WE HAVE TO GO DEEPER

Priya and I spent the evening (re)investigating various hypotheses why inception was failing. We discovered some new pieces of information: (1) changing the network to run on real images instead of random data doesn't help precision (both Priya and Edward reproduced this); (2) Caffe2 CPU convolution behaves differently than PyTorch convolution; this should be investigated; (3) even though Caffe2 Inception net gives different numeric output, it still manages to classify images with similar precision to the PyTorch version (Caffe2 89.844 precision@1, versus PyTorch 90.234 precision@1); (4) divergence in numerics doesn't happen until at the Mixed_5b layer. We still don't understand why the Conv2d weights to a normal distribution causes things to converge.

SOME META LESSONS

While working on trying the caffe2 conversion code in a different setting outside of the test suite (because there was pre-existing validation code), I noticed a few rough edges that we should deal with for the release.

(1) Our tracer is only able to trace a model once; if you trace it again it triggers an assert failure. A repro is in https://github.com/ezyang/pytorch/issues/142

(2) You can't trace models that use DataParallel, because this makes use of Scatter/Gather ops for which we don't have primspecs (this is unlikely to be profitable to trace, since the structure will depend on the number of GPUs on your machine.) But then you have to annoyingly hack it out. Would be nice if this was handled transparently.

(3) It is probably worth adding primspecs for the ops necessitated by  x[:, 0] = x[:, 0] * (0.229 / 0.5) + (0.485 - 0.5) / 0.5 (which occurs in Inception). Lack of this meant we had to tip toe around normalization

**Comment by ezyang** [comment]
I'm closing this since we're not going to merge it, and we're not looking at inception in the near future
